core/client/fs/src/main/java/alluxio/client/file/cache/PageId.java

```diff
@@ -23,14 +26,25 @@
 public class PageId {
   private final String mFileId;
   private final long mPageIndex;
+  private final Optional<CacheContext> mCacheContext;
```

Comment:
why do we include CacheContext in PageId?

Reply:
Good question, I also struggled with this for a while.  I'm just thinking each page could contains a cacheContext, but I didn't see a data structure for the page.  I include the cacheContext in PageId, so that the cacheContext can be accessed much easier in all places where the page is operated(put/get/evict), especially when we implement TTL or some other eviction policy later.
 
If we don't include the CacheContext in PageId, we might need to modify the signature of the get and put methods to pass the cacheContext into CacheManager.  What do you think?

Reply:
I am good to put CacheCotext into Put/Get signature ---- I think that's the correct

Reply:
Done.

Hi @dbw9580, I changed the signature of the get/put method in CacheManager,  it would be easier to get a cacheContext, but I guess it might cause lots of conflict with your local change.  Sorry for that :(